packages/modules/cart/src/models/cart.ts

```diff
@@ -1,181 +1,76 @@
+        foreignKey: true,
+      })
+      .nullable(),
+    items: model.hasMany(() => LineItem, {
+      mappedBy: "cart",
+    }),
+    shipping_methods: model.hasMany(() => ShippingMethod, {
+      mappedBy: "cart",
+    }),
   })
-  shipping_address_id: string | null
-
-  @ManyToOne(() => Address, {
-    cascade: [Cascade.PERSIST],
-    nullable: true,
+  .cascades({
+    delete: [
+      "items",
+      "shipping_methods",
+      "shipping_address",
+      "billing_address",
+    ],
   })
-  shipping_address: Rel<Address> | null
-
-  @BillingAddressIdIndex()
-  @ManyToOne({
-    entity: () => Address,
-    columnType: "text",
-    fieldName: "billing_address_id",
-    mapToPk: true,
-    nullable: true,
-  })
-  billing_address_id: string | null
-
-  @ManyToOne(() => Address, {
-    cascade: [Cascade.PERSIST],
-    nullable: true,
-  })
-  billing_address: Rel<Address> | null
-
-  @Property({ columnType: "jsonb", nullable: true })
-  metadata: Record<string, unknown> | null = null
-
-  @OneToMany(() => LineItem, (lineItem) => lineItem.cart, {
-    cascade: [Cascade.PERSIST, "soft-remove"] as any,
-  })
-  items = new Collection<Rel<LineItem>>(this)
-
-  @OneToMany(() => ShippingMethod, (shippingMethod) => shippingMethod.cart, {
-    cascade: [Cascade.PERSIST, "soft-remove"] as any,
-  })
-  shipping_methods = new Collection<Rel<ShippingMethod>>(this)
-
-  @Property({ columnType: "timestamptz", nullable: true })
-  completed_at: Date | null = null
-
-  @Property({
-    onCreate: () => new Date(),
-    columnType: "timestamptz",
-    defaultRaw: "now()",
-  })
-  created_at: Date
-
-  @Property({
-    onCreate: () => new Date(),
-    onUpdate: () => new Date(),
-    columnType: "timestamptz",
-    defaultRaw: "now()",
-  })
-  updated_at: Date
-
-  @DeletedAtIndex()
-  @Property({ columnType: "timestamptz", nullable: true })
-  deleted_at: Date | null = null
-
-  @BeforeCreate()
-  onCreate() {
-    this.id = generateEntityId(this.id, "cart")
-  }
-
-  @OnInit()
-  onInit() {
-    this.id = generateEntityId(this.id, "cart")
-  }
-}
+  .indexes([
+    {
+      name: "IDX_cart_region_id",
+      on: ["region_id"],
+      where: "deleted_at IS NULL AND region_id IS NOT NULL",
+    },
+    {
+      name: "IDX_cart_customer_id",
+      on: ["customer_id"],
+      where: "deleted_at IS NULL AND customer_id IS NOT NULL",
+    },
+    {
+      name: "IDX_cart_sales_channel_id",
+      on: ["sales_channel_id"],
+      where: "deleted_at IS NULL AND sales_channel_id IS NOT NULL",
+    },
+    {
+      name: "IDX_cart_curency_code",
+      on: ["currency_code"],
+      where: "deleted_at IS NULL",
+    },
+    {
+      name: "IDX_cart_shipping_address_id",
+      on: ["shipping_address_id" as any],
+      where: "deleted_at IS NULL AND shipping_address_id IS NOT NULL",
+    },
+    {
+      name: "IDX_cart_billing_address_id",
+      on: ["billing_address_id" as any],
+      where: "deleted_at IS NULL AND billing_address_id IS NOT NULL",
```

Comment:
Will have to remove `any` from the FK reference. But I will do that once we agree that this approach of `hasOne` with `foreignKey` is something we want to do